lib/fabrik/CfPlatformManager.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const Promise = require('bluebird');
+const BasePlatformManager = require('./BasePlatformManager');
+const utils = require('../utils');
+const assert = require('assert');
+const errors = require('../errors');
+const cloudController = require('../cf').cloudController;
+const logger = require('../logger');
+const SecurityGroupNotCreated = errors.SecurityGroupNotCreated;
+const SecurityGroupNotFound = errors.SecurityGroupNotFound;
+const NotImplemented = errors.NotImplemented;
+const ordinals = ['First', 'Second', 'Third', 'Fourth', 'Fifth', 'Sixth', 'Seventh', 'Eighth', 'Ninth', 'Tenth'];
+
+class CfPlatformManager extends BasePlatformManager {
+  constructor(guid, manager, context) {
+    super(guid, manager, context);
+    this.space_guid = context.space_guid;
+    this.cloudController = cloudController;
+  }
+
+  preInstanceProvisionOperations(options) {
+    /* jshint unused:false */
+    throw new NotImplemented('preInstanceProvisionOperations');
+  }
+
+  postInstanceProvisionOperations(options) {
+    return this.createSecurityGroup(this.space_guid, options.ruleId, options.ipRules);
+  }
+
+  preInstanceDeleteOperations(options) {
+    return this.deleteSecurityGroup(options.ruleId);
+  }
+
+  postInstanceUpdateOperations(options) {
+    return this.ensureSecurityGroupExists(this.space_guid, options.ruleId);
+  }
+
+  createSecurityGroup(space_guid, name, rules) {
+    logger.info(`Creating security group '${name}' with rules ...`, rules);
+    return utils
+      .retry(tries => {
+        logger.info(`+-> ${ordinals[tries]} attempt to create security group '${name}'...`);
+        return this.cloudController
+          .createSecurityGroup(name, rules, [space_guid])
+          .catch(err => {
+            logger.error(err);
+            throw err;
+          });
+      }, {
+        maxAttempts: 4,
+        minDelay: 1000
+      })
+      .then(securityGroup => securityGroup.metadata.guid)
+      .tap(guid => logger.info(`+-> Created security group with guid '${guid}'`))
+      .catch(err => {
+        logger.error(`+-> Failed to create security group ${name}`);
+        logger.error(err);
+        throw new SecurityGroupNotCreated(name);
+      });
+  }
+
+  ensureSecurityGroupExists(space_guid, name) {
```

Comment:
platformanager has space_guid already set, so you want to revisit this method signature?